src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final String WRONG_INDEX_MSG = "Invalid index. Check the list size";
+    public static final String NO_SUCH_ELEMENT = "Can't find such element";
+    public static final int INITIAL_SIZE = 10;
+
+    private T[] array = (T[]) new Object[INITIAL_SIZE];
+    private int currentSize = 0;
+    private int maxSize = INITIAL_SIZE;
+
     @Override
     public void add(T value) {
-
+        sizeCheck();
+        array[currentSize] = value;
+        currentSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > currentSize) {
+            throw new ArrayListIndexOutOfBoundsException(WRONG_INDEX_MSG);
+        }
+        sizeCheck();
+        if (index <= currentSize) {
+            T[] buffer = Arrays.copyOfRange(array, index, currentSize);
+            array[index] = value;
+            System.arraycopy(buffer, 0, array, index + 1, buffer.length);
+            currentSize++;
+        }
```

Comment:
![image](https://user-images.githubusercontent.com/58032907/195316361-a901b7f3-8346-4277-ad66-41dcd2565d8b.png)
also if-check here redundant, because we already checked this in previous if